src/Build/BackEnd/Components/RequestBuilder/IntrinsicTasks/MSBuild.cs

```diff
@@ -324,7 +326,21 @@ public async Task<bool> ExecuteInternal()
                     break;
                 }
 
-                if (FileSystems.Default.FileExists(projectPath) || (_skipNonexistentProjects == SkipNonexistentProjectsBehavior.Build))
+                // Try to get the behavior from metadata if it is undefined.
+                var skipNonexistPropjects = _skipNonexistentProjects;
```

Comment:
```suggestion
                var skipNonexistProjects = _skipNonexistentProjects;
```

(and in the other clone, if you don't just undo that.)